backend/views/retrieve_test_view.py

```diff
@@ -79,85 +101,172 @@ def retrieve_json_from_name_date(test_name, query_date_time, request_type):
+        return_dict["questions"] = question_map
         return return_dict
 
     # if it is not one of the above, then return nothing
     return {}
 
 
+def add_to_map(child_type, child_language_map, language_map):
+    if child_type in SINGLE_RETURN:
+        language_map[child_type] = child_language_map
+    elif child_type in language_map.keys():
+        language_map[child_type].append(child_language_map)
+    else:
+        language_map[child_type] = [child_language_map]
+    return language_map
+
+
+def get_items_map(parent_item, item_type_map, question_type_map, query_date_time,
+                  en_id, fr_id, children_map):
+    en_map, fr_map = get_items(parent_item, item_type_map, question_type_map, query_date_time,
+                               en_id, fr_id, children_map)
+    return {EN: en_map, FR: fr_map}
+
+
+def get_items(parent_item, item_type_map, question_type_map, query_date_time,
+              en_id, fr_id, children_map):
+    # get the parent id, get the type to determine how to handle it
+    parent_id, parent_type = get_item_type(
+        parent_item, item_type_map, question_type_map, query_date_time)
+    try:
+        children_types = children_map[parent_type]
+    except KeyError:
+        # if there the parent type does not have any children type to look up
+        # simply return the text_detail
+        return get_text_detail(parent_id, en_id, query_date_time), get_text_detail(parent_id, fr_id, query_date_time)
+    # otherwise, initialize return maps
+    en_map = {}
+    fr_map = {}
+    # a map to track the current id for a given child_type
+    # this also ensures that the id/order is sequential
+    order_map = {}
+    # get all items with parent_id
+    children_items = get_items_by_parent_id(parent_id, query_date_time)
+    # for each child
+    for child in children_items:
+        _, child_type = get_item_type(
+            child, item_type_map, question_type_map, query_date_time)
+        # if the type is in the list of children to return
+        if child_type in children_types:
+            # get the maps of their children
+            child_en, child_fr = get_items(
+                child, item_type_map, question_type_map, query_date_time,
+                en_id, fr_id, children_map)
+            # if they are dicts, add the id as a key/value pair
+            if isinstance(child_en, dict):
+                if child_type in order_map.keys():
+                    order_map[child_type] += 1
+                else:
+                    order_map[child_type] = 0
+                child_id = order_map[child_type]
+                child_en["id"] = child_id
+                child_fr["id"] = child_id
+            # add to the return map
+            en_map = add_to_map(child_type, child_en, en_map)
+            fr_map = add_to_map(child_type, child_fr, fr_map)
+    return en_map, fr_map
+
+# get the item id and item_type; or, if the item_type is question, get the question_type
+

```

Comment:
Remove white space

Reply:
removed whitespace for a few functions in 2235ae3